tests/components/progettihwsw/test_config_flow.py

```diff
@@ -0,0 +1,168 @@
+"""Test the ProgettiHWSW Automation config flow."""
+from homeassistant import config_entries, setup
+from homeassistant.components.progettihwsw.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.data_entry_flow import RESULT_TYPE_CREATE_ENTRY, RESULT_TYPE_FORM
+
+from tests.async_mock import patch
+
+mock_value_step_user = {
+    "title": "1R & 1IN Board",
+    "relay_count": 1,
+    "input_count": 1,
+    "is_old": False,
+}
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)